packages/renderer/src/lib/preferences/PreferencesRendering.svelte

```diff
@@ -12,35 +12,39 @@ import PreferencesRenderingItem from './PreferencesRenderingItem.svelte';
 import SettingsPage from './SettingsPage.svelte';
 import { isDefaultScope, isPropertyValidInContext } from './Util';
 
-export let properties: IConfigurationPropertyRecordedSchema[] = [];
-export let key: string;
+interface Props {
+  properties: IConfigurationPropertyRecordedSchema[];
+  key: string;
+  searchValue?: string;
+}
+
+let { properties = [], key, searchValue = '' }: Props = $props();
 
 // Context variables
 let contextsUnsubscribe: Unsubscriber;
 let globalContext: ContextUI;
 
 // Search and matching records
 let updateSearchValueTimeout: NodeJS.Timeout;
-let matchingRecords: Map<string, IConfigurationPropertyRecordedSchema[]>;
-export let searchValue = '';
-$: searchValue;
-$: matchingRecords = properties
-  .filter(property => property.parentId.startsWith(key) && isDefaultScope(property.scope) && !property.hidden)
-  .filter(property => isPropertyValidInContext(property.when, globalContext))
-  .filter(
-    property =>
-      !searchValue ||
-      matchValue(property.title, searchValue) ||
-      (!!property.description && matchValue(property.description, searchValue)) ||
-      (!!property.markdownDescription && matchValue(property.markdownDescription, searchValue)),
-  )
-  .reduce((map, property) => {
-    if (!map.has(property.parentId)) {
-      map.set(property.parentId, []);
-    }
-    map.get(property.parentId)?.push(property);
-    return map;
-  }, new Map<string, IConfigurationPropertyRecordedSchema[]>());
+const matchingRecords = $derived(
+  properties
+    .filter(property => property.parentId.startsWith(key) && isDefaultScope(property.scope) && !property.hidden)
+    .filter(property => isPropertyValidInContext(property.when, globalContext))
+    .filter(
+      property =>
+        !searchValue ||
+        matchValue(property.title, searchValue) ||
+        (!!property.description && matchValue(property.description, searchValue)) ||
+        (!!property.markdownDescription && matchValue(property.markdownDescription, searchValue)),
+    )
+    .reduce((map, property) => {
+      if (!map.has(property.parentId)) {
+        map.set(property.parentId, []);
+      }
+      map.get(property.parentId)?.push(property);
+      return map;
+    }, new Map<string, IConfigurationPropertyRecordedSchema[]>()),
```

Comment:
(out of scope of this PR), but this is a bit of a performance issue, as we do O(n*4), I added a comment in the code guidelines https://github.com/podman-desktop/podman-desktop/issues/9821#issuecomment-2567478281